ui/mirage/models/variable.ts

```diff
@@ -0,0 +1,20 @@
+import { Model, belongsTo, hasMany } from 'ember-cli-mirage';
+import { Variable, Ref } from 'waypoint-pb';
+
+export default Model.extend({
+  project: belongsTo(),
+
+  toProtobuf(): Variable {
+    let result = new Variable();
+
+    result.setServer();
+    result.setName(this.name);
+    if (this.hcl) {
+      result.setHcl(this.hcl);
+    } else {
+      result.setStr(this.str);
+    }
```

Comment:
Does this need to same “unset the other field” behavior we see above?

```suggestion
    if (this.hcl) {
      result.setHcl(this.hcl);
      result.setStr('');
    } else {
      result.setHcl('');
      result.setStr(this.str);
    }
```

(aside: this aspect of protobuf `oneof` fields is a little inelegant isn’t it?)